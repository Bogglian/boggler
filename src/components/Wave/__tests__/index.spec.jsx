import React from 'react';
import { shallow } from 'enzyme';
import { Wave } from '..';

describe('Wave Component', () => {
  describe('user possibly create a Wave component.', () => {
    it('should render a Wave', () => {
      const component = shallow(<Wave />);

      expect(component).toMatchSnapshot();
    });
  });
});
