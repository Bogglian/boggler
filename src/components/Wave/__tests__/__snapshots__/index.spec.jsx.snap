// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wave Component user possibly create a Wave component. should render a Wave 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wave />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <Wrapper
          height="200"
          position={0}
          seconds={0}
          showPosition={true}
          width="1650"
        />
        <div
          className="player-wrapper"
        >
          <ReactPlayer
            config={
              Object {
                "dailymotion": Object {
                  "params": Object {
                    "api": 1,
                    "endscreen-enable": false,
                  },
                  "preload": false,
                },
                "facebook": Object {
                  "appId": "1309697205772819",
                },
                "file": Object {
                  "attributes": Object {},
                  "dashVersion": "2.9.2",
                  "forceAudio": false,
                  "forceDASH": false,
                  "forceHLS": false,
                  "forceVideo": false,
                  "hlsOptions": Object {},
                  "hlsVersion": "0.10.1",
                  "tracks": Array [],
                },
                "mixcloud": Object {
                  "options": Object {
                    "hide_cover": 1,
                  },
                },
                "soundcloud": Object {
                  "options": Object {
                    "buying": false,
                    "download": false,
                    "liking": false,
                    "sharing": false,
                    "show_comments": false,
                    "show_playcount": false,
                    "visual": true,
                  },
                },
                "twitch": Object {
                  "options": Object {},
                },
                "vimeo": Object {
                  "playerOptions": Object {
                    "autopause": false,
                    "byline": false,
                    "portrait": false,
                    "title": false,
                  },
                  "preload": false,
                },
                "wistia": Object {
                  "options": Object {},
                },
                "youtube": Object {
                  "playerVars": Object {
                    "iv_load_policy": 3,
                    "modestbranding": 1,
                    "playsinline": 1,
                    "rel": 0,
                    "showinfo": 0,
                  },
                  "preload": false,
                },
              }
            }
            controls={true}
            height="100%"
            light={false}
            loop={false}
            muted={false}
            onBuffer={[Function]}
            onDisablePIP={[Function]}
            onDuration={[Function]}
            onEnablePIP={[Function]}
            onEnded={[Function]}
            onError={[Function]}
            onPause={[Function]}
            onPlay={[Function]}
            onProgress={[Function]}
            onReady={[Function]}
            onSeek={[Function]}
            onStart={[Function]}
            pip={false}
            playbackRate={1}
            playing={false}
            playsinline={false}
            progressInterval={70}
            style={Object {}}
            volume={null}
            width="100%"
            wrapper="div"
          />
        </div>
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Wrapper
            height="200"
            position={0}
            seconds={0}
            showPosition={true}
            width="1650"
          />,
          <div
            className="player-wrapper"
          >
            <ReactPlayer
              config={
                Object {
                  "dailymotion": Object {
                    "params": Object {
                      "api": 1,
                      "endscreen-enable": false,
                    },
                    "preload": false,
                  },
                  "facebook": Object {
                    "appId": "1309697205772819",
                  },
                  "file": Object {
                    "attributes": Object {},
                    "dashVersion": "2.9.2",
                    "forceAudio": false,
                    "forceDASH": false,
                    "forceHLS": false,
                    "forceVideo": false,
                    "hlsOptions": Object {},
                    "hlsVersion": "0.10.1",
                    "tracks": Array [],
                  },
                  "mixcloud": Object {
                    "options": Object {
                      "hide_cover": 1,
                    },
                  },
                  "soundcloud": Object {
                    "options": Object {
                      "buying": false,
                      "download": false,
                      "liking": false,
                      "sharing": false,
                      "show_comments": false,
                      "show_playcount": false,
                      "visual": true,
                    },
                  },
                  "twitch": Object {
                    "options": Object {},
                  },
                  "vimeo": Object {
                    "playerOptions": Object {
                      "autopause": false,
                      "byline": false,
                      "portrait": false,
                      "title": false,
                    },
                    "preload": false,
                  },
                  "wistia": Object {
                    "options": Object {},
                  },
                  "youtube": Object {
                    "playerVars": Object {
                      "iv_load_policy": 3,
                      "modestbranding": 1,
                      "playsinline": 1,
                      "rel": 0,
                      "showinfo": 0,
                    },
                    "preload": false,
                  },
                }
              }
              controls={true}
              height="100%"
              light={false}
              loop={false}
              muted={false}
              onBuffer={[Function]}
              onDisablePIP={[Function]}
              onDuration={[Function]}
              onEnablePIP={[Function]}
              onEnded={[Function]}
              onError={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onProgress={[Function]}
              onReady={[Function]}
              onSeek={[Function]}
              onStart={[Function]}
              pip={false}
              playbackRate={1}
              playing={false}
              playsinline={false}
              progressInterval={70}
              style={Object {}}
              volume={null}
              width="100%"
              wrapper="div"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buffer": undefined,
            "height": "200",
            "position": 0,
            "seconds": 0,
            "showPosition": true,
            "width": "1650",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ReactPlayer
              config={
                Object {
                  "dailymotion": Object {
                    "params": Object {
                      "api": 1,
                      "endscreen-enable": false,
                    },
                    "preload": false,
                  },
                  "facebook": Object {
                    "appId": "1309697205772819",
                  },
                  "file": Object {
                    "attributes": Object {},
                    "dashVersion": "2.9.2",
                    "forceAudio": false,
                    "forceDASH": false,
                    "forceHLS": false,
                    "forceVideo": false,
                    "hlsOptions": Object {},
                    "hlsVersion": "0.10.1",
                    "tracks": Array [],
                  },
                  "mixcloud": Object {
                    "options": Object {
                      "hide_cover": 1,
                    },
                  },
                  "soundcloud": Object {
                    "options": Object {
                      "buying": false,
                      "download": false,
                      "liking": false,
                      "sharing": false,
                      "show_comments": false,
                      "show_playcount": false,
                      "visual": true,
                    },
                  },
                  "twitch": Object {
                    "options": Object {},
                  },
                  "vimeo": Object {
                    "playerOptions": Object {
                      "autopause": false,
                      "byline": false,
                      "portrait": false,
                      "title": false,
                    },
                    "preload": false,
                  },
                  "wistia": Object {
                    "options": Object {},
                  },
                  "youtube": Object {
                    "playerVars": Object {
                      "iv_load_policy": 3,
                      "modestbranding": 1,
                      "playsinline": 1,
                      "rel": 0,
                      "showinfo": 0,
                    },
                    "preload": false,
                  },
                }
              }
              controls={true}
              height="100%"
              light={false}
              loop={false}
              muted={false}
              onBuffer={[Function]}
              onDisablePIP={[Function]}
              onDuration={[Function]}
              onEnablePIP={[Function]}
              onEnded={[Function]}
              onError={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onProgress={[Function]}
              onReady={[Function]}
              onSeek={[Function]}
              onStart={[Function]}
              pip={false}
              playbackRate={1}
              playing={false}
              playsinline={false}
              progressInterval={70}
              style={Object {}}
              volume={null}
              width="100%"
              wrapper="div"
            />,
            "className": "player-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "config": Object {
                "dailymotion": Object {
                  "params": Object {
                    "api": 1,
                    "endscreen-enable": false,
                  },
                  "preload": false,
                },
                "facebook": Object {
                  "appId": "1309697205772819",
                },
                "file": Object {
                  "attributes": Object {},
                  "dashVersion": "2.9.2",
                  "forceAudio": false,
                  "forceDASH": false,
                  "forceHLS": false,
                  "forceVideo": false,
                  "hlsOptions": Object {},
                  "hlsVersion": "0.10.1",
                  "tracks": Array [],
                },
                "mixcloud": Object {
                  "options": Object {
                    "hide_cover": 1,
                  },
                },
                "soundcloud": Object {
                  "options": Object {
                    "buying": false,
                    "download": false,
                    "liking": false,
                    "sharing": false,
                    "show_comments": false,
                    "show_playcount": false,
                    "visual": true,
                  },
                },
                "twitch": Object {
                  "options": Object {},
                },
                "vimeo": Object {
                  "playerOptions": Object {
                    "autopause": false,
                    "byline": false,
                    "portrait": false,
                    "title": false,
                  },
                  "preload": false,
                },
                "wistia": Object {
                  "options": Object {},
                },
                "youtube": Object {
                  "playerVars": Object {
                    "iv_load_policy": 3,
                    "modestbranding": 1,
                    "playsinline": 1,
                    "rel": 0,
                    "showinfo": 0,
                  },
                  "preload": false,
                },
              },
              "controls": true,
              "height": "100%",
              "light": false,
              "loop": false,
              "muted": false,
              "onBuffer": [Function],
              "onDisablePIP": [Function],
              "onDuration": [Function],
              "onEnablePIP": [Function],
              "onEnded": [Function],
              "onError": [Function],
              "onPause": [Function],
              "onPlay": [Function],
              "onProgress": [Function],
              "onReady": [Function],
              "onSeek": [Function],
              "onStart": [Function],
              "pip": false,
              "playbackRate": 1,
              "playing": false,
              "playsinline": false,
              "progressInterval": 70,
              "style": Object {},
              "url": undefined,
              "volume": null,
              "width": "100%",
              "wrapper": "div",
            },
            "ref": [Function],
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htoDjs",
          "isStatic": true,
          "rules": Array [
            "
  & {
    width: 100%;
    height: 100%;
    display: block;
    
  }
  
  .player-wrapper {
    height: 3%;
    width: 97%;
    position : absolute;
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htoDjs",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gzVnrw",
        "isStatic": true,
        "rules": Array [
          "
  & {
    width: 100%;
    height: 100%;
    padding: 1%;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-gzVnrw",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <Wrapper
            height="200"
            position={0}
            seconds={0}
            showPosition={true}
            width="1650"
          />
          <div
            className="player-wrapper"
          >
            <ReactPlayer
              config={
                Object {
                  "dailymotion": Object {
                    "params": Object {
                      "api": 1,
                      "endscreen-enable": false,
                    },
                    "preload": false,
                  },
                  "facebook": Object {
                    "appId": "1309697205772819",
                  },
                  "file": Object {
                    "attributes": Object {},
                    "dashVersion": "2.9.2",
                    "forceAudio": false,
                    "forceDASH": false,
                    "forceHLS": false,
                    "forceVideo": false,
                    "hlsOptions": Object {},
                    "hlsVersion": "0.10.1",
                    "tracks": Array [],
                  },
                  "mixcloud": Object {
                    "options": Object {
                      "hide_cover": 1,
                    },
                  },
                  "soundcloud": Object {
                    "options": Object {
                      "buying": false,
                      "download": false,
                      "liking": false,
                      "sharing": false,
                      "show_comments": false,
                      "show_playcount": false,
                      "visual": true,
                    },
                  },
                  "twitch": Object {
                    "options": Object {},
                  },
                  "vimeo": Object {
                    "playerOptions": Object {
                      "autopause": false,
                      "byline": false,
                      "portrait": false,
                      "title": false,
                    },
                    "preload": false,
                  },
                  "wistia": Object {
                    "options": Object {},
                  },
                  "youtube": Object {
                    "playerVars": Object {
                      "iv_load_policy": 3,
                      "modestbranding": 1,
                      "playsinline": 1,
                      "rel": 0,
                      "showinfo": 0,
                    },
                    "preload": false,
                  },
                }
              }
              controls={true}
              height="100%"
              light={false}
              loop={false}
              muted={false}
              onBuffer={[Function]}
              onDisablePIP={[Function]}
              onDuration={[Function]}
              onEnablePIP={[Function]}
              onEnded={[Function]}
              onError={[Function]}
              onPause={[Function]}
              onPlay={[Function]}
              onProgress={[Function]}
              onReady={[Function]}
              onSeek={[Function]}
              onStart={[Function]}
              pip={false}
              playbackRate={1}
              playing={false}
              playsinline={false}
              progressInterval={70}
              style={Object {}}
              volume={null}
              width="100%"
              wrapper="div"
            />
          </div>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Wrapper
              height="200"
              position={0}
              seconds={0}
              showPosition={true}
              width="1650"
            />,
            <div
              className="player-wrapper"
            >
              <ReactPlayer
                config={
                  Object {
                    "dailymotion": Object {
                      "params": Object {
                        "api": 1,
                        "endscreen-enable": false,
                      },
                      "preload": false,
                    },
                    "facebook": Object {
                      "appId": "1309697205772819",
                    },
                    "file": Object {
                      "attributes": Object {},
                      "dashVersion": "2.9.2",
                      "forceAudio": false,
                      "forceDASH": false,
                      "forceHLS": false,
                      "forceVideo": false,
                      "hlsOptions": Object {},
                      "hlsVersion": "0.10.1",
                      "tracks": Array [],
                    },
                    "mixcloud": Object {
                      "options": Object {
                        "hide_cover": 1,
                      },
                    },
                    "soundcloud": Object {
                      "options": Object {
                        "buying": false,
                        "download": false,
                        "liking": false,
                        "sharing": false,
                        "show_comments": false,
                        "show_playcount": false,
                        "visual": true,
                      },
                    },
                    "twitch": Object {
                      "options": Object {},
                    },
                    "vimeo": Object {
                      "playerOptions": Object {
                        "autopause": false,
                        "byline": false,
                        "portrait": false,
                        "title": false,
                      },
                      "preload": false,
                    },
                    "wistia": Object {
                      "options": Object {},
                    },
                    "youtube": Object {
                      "playerVars": Object {
                        "iv_load_policy": 3,
                        "modestbranding": 1,
                        "playsinline": 1,
                        "rel": 0,
                        "showinfo": 0,
                      },
                      "preload": false,
                    },
                  }
                }
                controls={true}
                height="100%"
                light={false}
                loop={false}
                muted={false}
                onBuffer={[Function]}
                onDisablePIP={[Function]}
                onDuration={[Function]}
                onEnablePIP={[Function]}
                onEnded={[Function]}
                onError={[Function]}
                onPause={[Function]}
                onPlay={[Function]}
                onProgress={[Function]}
                onReady={[Function]}
                onSeek={[Function]}
                onStart={[Function]}
                pip={false}
                playbackRate={1}
                playing={false}
                playsinline={false}
                progressInterval={70}
                style={Object {}}
                volume={null}
                width="100%"
                wrapper="div"
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "buffer": undefined,
              "height": "200",
              "position": 0,
              "seconds": 0,
              "showPosition": true,
              "width": "1650",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ReactPlayer
                config={
                  Object {
                    "dailymotion": Object {
                      "params": Object {
                        "api": 1,
                        "endscreen-enable": false,
                      },
                      "preload": false,
                    },
                    "facebook": Object {
                      "appId": "1309697205772819",
                    },
                    "file": Object {
                      "attributes": Object {},
                      "dashVersion": "2.9.2",
                      "forceAudio": false,
                      "forceDASH": false,
                      "forceHLS": false,
                      "forceVideo": false,
                      "hlsOptions": Object {},
                      "hlsVersion": "0.10.1",
                      "tracks": Array [],
                    },
                    "mixcloud": Object {
                      "options": Object {
                        "hide_cover": 1,
                      },
                    },
                    "soundcloud": Object {
                      "options": Object {
                        "buying": false,
                        "download": false,
                        "liking": false,
                        "sharing": false,
                        "show_comments": false,
                        "show_playcount": false,
                        "visual": true,
                      },
                    },
                    "twitch": Object {
                      "options": Object {},
                    },
                    "vimeo": Object {
                      "playerOptions": Object {
                        "autopause": false,
                        "byline": false,
                        "portrait": false,
                        "title": false,
                      },
                      "preload": false,
                    },
                    "wistia": Object {
                      "options": Object {},
                    },
                    "youtube": Object {
                      "playerVars": Object {
                        "iv_load_policy": 3,
                        "modestbranding": 1,
                        "playsinline": 1,
                        "rel": 0,
                        "showinfo": 0,
                      },
                      "preload": false,
                    },
                  }
                }
                controls={true}
                height="100%"
                light={false}
                loop={false}
                muted={false}
                onBuffer={[Function]}
                onDisablePIP={[Function]}
                onDuration={[Function]}
                onEnablePIP={[Function]}
                onEnded={[Function]}
                onError={[Function]}
                onPause={[Function]}
                onPlay={[Function]}
                onProgress={[Function]}
                onReady={[Function]}
                onSeek={[Function]}
                onStart={[Function]}
                pip={false}
                playbackRate={1}
                playing={false}
                playsinline={false}
                progressInterval={70}
                style={Object {}}
                volume={null}
                width="100%"
                wrapper="div"
              />,
              "className": "player-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "config": Object {
                  "dailymotion": Object {
                    "params": Object {
                      "api": 1,
                      "endscreen-enable": false,
                    },
                    "preload": false,
                  },
                  "facebook": Object {
                    "appId": "1309697205772819",
                  },
                  "file": Object {
                    "attributes": Object {},
                    "dashVersion": "2.9.2",
                    "forceAudio": false,
                    "forceDASH": false,
                    "forceHLS": false,
                    "forceVideo": false,
                    "hlsOptions": Object {},
                    "hlsVersion": "0.10.1",
                    "tracks": Array [],
                  },
                  "mixcloud": Object {
                    "options": Object {
                      "hide_cover": 1,
                    },
                  },
                  "soundcloud": Object {
                    "options": Object {
                      "buying": false,
                      "download": false,
                      "liking": false,
                      "sharing": false,
                      "show_comments": false,
                      "show_playcount": false,
                      "visual": true,
                    },
                  },
                  "twitch": Object {
                    "options": Object {},
                  },
                  "vimeo": Object {
                    "playerOptions": Object {
                      "autopause": false,
                      "byline": false,
                      "portrait": false,
                      "title": false,
                    },
                    "preload": false,
                  },
                  "wistia": Object {
                    "options": Object {},
                  },
                  "youtube": Object {
                    "playerVars": Object {
                      "iv_load_policy": 3,
                      "modestbranding": 1,
                      "playsinline": 1,
                      "rel": 0,
                      "showinfo": 0,
                    },
                    "preload": false,
                  },
                },
                "controls": true,
                "height": "100%",
                "light": false,
                "loop": false,
                "muted": false,
                "onBuffer": [Function],
                "onDisablePIP": [Function],
                "onDuration": [Function],
                "onEnablePIP": [Function],
                "onEnded": [Function],
                "onError": [Function],
                "onPause": [Function],
                "onPlay": [Function],
                "onProgress": [Function],
                "onReady": [Function],
                "onSeek": [Function],
                "onStart": [Function],
                "pip": false,
                "playbackRate": 1,
                "playing": false,
                "playsinline": false,
                "progressInterval": 70,
                "style": Object {},
                "url": undefined,
                "volume": null,
                "width": "100%",
                "wrapper": "div",
              },
              "ref": [Function],
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htoDjs",
            "isStatic": true,
            "rules": Array [
              "
  & {
    width: 100%;
    height: 100%;
    display: block;
    
  }
  
  .player-wrapper {
    height: 3%;
    width: 97%;
    position : absolute;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htoDjs",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-gzVnrw",
          "isStatic": true,
          "rules": Array [
            "
  & {
    width: 100%;
    height: 100%;
    padding: 1%;
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-gzVnrw",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
